# Homework 6, Alex Pine (akp258@nyu.edu)

In [2]:
import pandas

filename = '/Users/pinesol/causal/hw6/late.csv'
df = pandas.read_csv(filename)

### Problem 1.i: Compute $\delta_{IV}$

In [54]:
import numpy as np

delta_iv = (np.cov(df['Y'], df['Z'], bias=True)[0][1] / 
            np.cov(df['D'], df['Z'], bias=True)[0][1])
print delta_iv

1.93517078741


$\delta_{IV}$ equals 1.94.

### Problem 1.ii: Find $P_A$, $P_N$, and $P_C$.

$$ P_A = E[D^* \mid Z = 0]$$

In [25]:
p_always = np.mean(df[df['Z'] == 0]['D'])
p_always

0.31320504313205044

The probability of being an always-taker is 0.31.

$$P_N = 1 - E[D^* \mid Z = 1]$$

In [26]:
p_never = 1 - np.mean(df[df['Z'] == 1]['D'])
p_never

0.32558139534883723

The probability of being an never-taker is 0.33.

$$P_C = 1 - P_A - P_N$$

In [28]:
p_complier = 1 - p_always - p_never
p_complier

0.36121356151911233

The probability of being a compiler is 0.36.

### Problem 1.iii: Find $E[Y^1 \mid A]$ and $E[Y^0 \mid N]$

$$ E[Y^1 \mid A] = E[Y^* \mid Z = 0, D = 1] $$

In [50]:
treated_always = np.mean(df[(df['Z'] == 0) & (df['D'] == 1)]['Y'])
treated_always

103.0957631107627

The expected value of $Y^1$ on the always-takers is 103.10.

$$ E[Y^0 \mid N] = E[Y^* \mid Z = 1, D = 0] $$

In [49]:
untreated_never = np.mean(df[(df['Z'] == 1) & (df['D'] == 0)]['Y'])
untreated_never

98.84696178636905

The expected value of $Y^0$ on the never-takers is 98.85.

### 1.iv: Compute $\delta_{LATE}$

Using the values for $E[Y^1 \mid C]$ and $E[Y^0 \mid C]$ from slide 26 of lecture 7.

In [55]:
## slide 26 formula

e_y_z1d1 = np.mean(df[(df['Z'] == 1) & (df['D'] == 1)]['Y'])

treated_comps = e_y_z1d1 + (p_always/p_complier)*(e_y_z1d1-treated_always)
e_y_z0d0 = np.mean(df[(df['Z'] == 0) & (df['D'] == 0)]['Y'])
untreated_comps = e_y_z0d0 + (p_never/p_complier)*(e_y_z0d0-untreated_never)

late = treated_comps - untreated_comps
late

1.9351707874081399

The local average treatment effect is 1.94.

The linear IV estimator for the average treatment effect is nearly the exact same as that of the local average treatment effect. 